atomicapp/nulecule/main.py

```diff
@@ -183,6 +184,11 @@ def run(self, cli_provider, answers_output, ask,
         # it does exist it will be just be loaded and returned
         self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
 
+        # If no provider is specified within answers.conf or --provider=
+        # we should try and get a default one from the artifacts folder
+        if self.answers[GLOBAL_CONF][PROVIDER_KEY] is None and cli_provider is None:
+            self.answers[GLOBAL_CONF][PROVIDER_KEY] = Utils.getDefaultArtifact(self.app_path)
+
```

Comment:
If you look at `getDefaultArtifact` it actually get's a default artifact by determining what's been provided in the `/artifacts` folder of the Nuleculized container. Wouldn't determining it that way be much more intuitive than checking to see if "kubernetes/DEFAULT_PROVIDER" has been set or not?